### ISSUE TITLE
Toggle fullscreen problem

### ISSUE THREAD

**Issue by region51** [mpv-player/mpv#2932]
Toggle fullscreen spasm, twitch
```
[maxim@ip200-37 ~]$ mpv -V
mpv 0.16.0 (C) 2000-2016 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
ffmpeg library versions:
   libavutil       54.31.100
   libavcodec      56.60.100
   libavformat     56.40.101
   libswscale      3.1.101
   libavfilter     5.40.101
   libswresample   1.2.101
ffmpeg version: 2.8.6

```
Video 3.05 MB https://www.dropbox.com/s/9l3y722yyns72bc/mpv-fs-bad.mkv?dl=0
Sorry for my bad English.

**Comment by region51** [comment]
ps. FullScreen Toggle memory leak??? Memory use climbs at a pretty fast rate.

**Comment by lachs0r** [comment]
That’s just your window manager. Memory leak could also be your drivers.

**Comment by region51** [comment]
@lachs0r 
>That’s just your window manager.

everywhere it's OK, it is only with mpv.

**Comment by region51** [comment]
>Memory leak could also be your drivers.

it is only with mpv.

**Comment by lachs0r** [comment]
On Friday, March 11, 2016 11:05:55 AM CET Maxim Orlov wrote:
> >Memory leak could also be your drivers.
> 
> it is only with mpv.

Doesn’t really matter. I’ve seen many Intel driver bugs that caused 
misbehavior in very few applications, mpv being one of them. I actually 
remember the same symptoms with Intel that you describe (memory leak in 
fullscreen mode), but this also plagued e.g. dolphin-emu.
This got fixed about a year ago. I don’t know whether that fix was in the 
kernel, Mesa, or the X11 driver, but it certainly wasn’t mpv’s fault.

**Comment by LateAdopter** [comment]
I have been using Doug McMahon's Trusty builds on my Braswell N3150 board for a few months and I get the same problem. Here is some more background, just for info:

For --vo=opengl or --vo=vaapi

Without "tearfree" "true" , I get tearing in MPV, but fullscreen doesn't get killed.

With tearfree and fullscreen together MPV gets killed in 18 seconds for out of memory.

With --vo=xv tearfree + fullscreen is OK

Windowed is OK

I have tried with with some of the Doug McMahon MPV builds from 0.8.3 to 0.16.0 with or without EGL. They all behave the same.

The N3150 is pretty feeble but it can, just about, do --vo=xv --vf=yadif=field with my 1080i50 recordings. I'm happy enough with that. The advantage is that it only takes 13 Watts wallplug power, while doing that, and no need for fans.

There's nothing to be gained by using --vo=opengl or --hwdec=vaapi until --vf=vavpp:deint=motion-compensated is working. All possible modes with -vf=yadif=field are about 65% of 2060MHz cpu load, whereas --hwdec=vaapi --vo=vaapi --vf=vavpp:deint=motion-compensated is about 8% of 1500MHz cpu load, but the pictures jump about.

My configuration is unusual, but Kodi 16.0 and Kodi 17 nightly, including the EGL/NV12 stuff and motion compensated deinterlace, all work properly.

My more or less stable configuration, which works well with Kodi 16.0 and Kodi 17 nightly:

TahrPup64 6.0.5
Kernel 4.4.4
DRM 2.4.64-1~ubuntu14.04.1_amd64
xserver-xorg-video-intel 2.99.917+git1601011930.ec92e2~gd~t_amd64
mesa-lts-wily (11.0.2)
Libva/Intel-vaapi-driver 1.6.2

There are no released drivers for Cherryview, and I have noticed a few commits tagged Cherryview recently. When they percolate as far as ~oibaf PPA, I will try them out and report back.

Thank you for your work (and Doug McMahon too)
